cpp/src/arrow/compute/row/row_encoder_internal.h

```diff
@@ -267,37 +272,44 @@ struct ARROW_EXPORT NullKeyEncoder : KeyEncoder {
 /// created by concatenating the encoded form of each column. The encoding
 /// for each column depends on its data type.
 ///
-/// This is used to encode columns into row-major format, which will be beneficial for grouping and joining operations.
+/// This is used to encode columns into row-major format, which will be
+/// beneficial for grouping and joining operations.
 ///
 /// Unlike DuckDB and arrow-rs, currently this row format can not help
 /// sortings because the row-format is uncomparable.
 ///
+/// # Key Column Encoding
+///
 /// The row format is composed of the the KeyColumn encodings for each,
 /// and the column is encoded as follows:
 /// 1. A null byte for each column, indicating whether the column is null.
 ///    "1" for null, "0" for non-null.
 /// 2. The "fixed width" encoding for the column, it would exist whether
 ///    the column is null or not.
-/// 3. The "variable width" encoding for the column, it would exists only
+/// 3. The "variable payload" encoding for the column, it would exists only
```

Comment:
Does it count into the "length" part for the var-length column (which will exist no matter the column is null or not)?

Reply:
I add some descr below